### ISSUE TITLE
Add OAuth authentication

### ISSUE THREAD

**Issue by gustavkarlsson** [gustavkarlsson/rocketchat-jira-trigger#4]
Plaintext passwords should still be supported though.

**Comment by gustavkarlsson** [comment]
Jira's java rest client doesn't yet support OAuth, so this would require some amount of hacking, possibly replacing the underlying http client.

**Comment by Eloo** [comment]
it seems that there is an example how oauth should be used
https://developer.atlassian.com/jiradev/jira-apis/jira-rest-apis/jira-rest-api-tutorials/jira-rest-api-example-oauth-authentication

would it be possible to integrate this?

**Comment by gustavkarlsson** [comment]
Yes. It's possible, although it would require some effort. This is definitely a feature I'm planning on adding at some point. Is this something you need @Eloo?

**Comment by Eloo** [comment]
@gustavkarlsson yeah would really great to have this implemented so we don't need to use a normal(tech) user.

**Comment by tobru** [comment]
I would love to see this as well. :+1:
